app.rb

```diff
@@ -0,0 +1,120 @@
+require_relative 'person'
+require_relative 'book'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'rental'
+
+class App
+  attr_accessor :person, :books, :rentals
+
+  def initialize()
+    @person = []
+    @books = []
+    @rentals = []
+  end
+
+  def create_book
+    puts ''
+    print 'Enter book title: '
+    title = gets.chomp
+    print 'Enter book author: '
+    author = gets.chomp
+    @books << Book.new(title, author)
+    puts 'Success'
+  end
+
+  def list_all_books
+    puts ''
+    @books.each { |book| puts "Title: #{book.title}", "Author: #{book.author}" }
+  end
+
+  def create_person
+    puts ''
+    print 'Enter 1 to add Student and 2 to add Teacher: '
+    selection = gets.chomp.to_i
+    print 'Enter name: '
+    name = gets.chomp
+    print 'Enter age: '
+    age = gets.chomp.to_i
+
+    case selection
+    when 1
+      print 'Does student have parent permission [Y/N]: '
+      permission = gets.chomp.downcase == 'y'
+      student = Student.new(1, age, permission, name)
+      @person << student
+
+    when 2
+      print 'What is the teachers specialization: '
+      specialization = gets.chomp
+      teacher = Teacher.new(age, specialization, name)
+      @person << teacher
+    end
+
+    print 'Success'
+  end
+
+  def list_all_people
+    puts ''
+    @person.each do |person|
+      puts ''
+      puts "[#{person.class}]"
```

Comment:
As in the [example video](https://www.youtube.com/watch?v=vkkgrhD6aXQ), it would be best if you included the IDs when listing all people. This makes it easy to check a person's ID in order to list their rentals. Although the IDs are included when creating a rental it is not an efficient way to check for people's IDs.

![image](https://github.com/seyioyemade/oop-school-library/assets/38088818/1f22e769-1a39-4437-9263-891f99c84ca5)

### OR

Since your UI displays the list of IDs when the view rentals option is selected, you can include the person's name next to each ID so that users know to whom the ID belongs.

For example:

![image](https://github.com/seyioyemade/oop-school-library/assets/38088818/697a5785-2a22-437e-a85c-3c15167c7e37)


Otherwise, everything in the Basic UI works as expected. Fantastic and well done!!! 👏🏾💯